### PR TITLE
Fixing the build and a few other things

### PR DESCRIPTION
**Pull request by joes-myob**
This Pull request does a few things.

1. Fixes the build by introducing a dummy test that is run - because there are no tests its showing an example of how to start building them in.
2. updates the buildkite configuration to use a more expected standard with buildkite. `.buildkite/pipeline.yml` file
3. This file uses the Buildkite docker integration to run the tests in a Golang container in CI
4. Adds a `Makefile` with the ability to run install / test and build binaries for target operating systems linux/windows/MacOS
5. Added some LDFLAGS for extra info in the binary of what branch and commit sha built it

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+agents:
+  queue: oss-lab
+
+steps:
+  - label: ":rocket: Testing.."
+    command: make install-deps test
+    plugins:
+      - docker#v3.9.0:
```
Comment (joes-myob) [inline .buildkite/pipeline.yml:8]: Buildkite plugin https://github.com/buildkite-plugins/docker-buildkite-plugin

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package main_test
+
+import "testing"
+
+func Test_For_CI(t *testing.T) {
```
Comment (joes-myob) [inline main_test.go:5]: This is just a random test to ensure its being executed in the pipeline

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+- Renamed `VERSION` file to `CHANGELOG.md`
```
Comment (PhonesMYOB) [inline CHANGELOG.md]: Hey @joes-myob , hubr looks for a `VERSION` file and uses that for releasing and bumping, if you rename it to CHANGELOG and change how it's organized some of hubr's functionality will break.

Comment (joes-myob) [reply CHANGELOG.md]: ah ha - thanks for pointing that out - can you link me to the code where it's looking for the VERSION file?

Comment (joes-myob) [reply CHANGELOG.md]: Looks like there is an arg to the path of the VERSION file
https://github.com/myob-oss/hubr/blob/master/main.go#L1389 - but I still cant see _where_ the code is looking for a "VERSION" file? @PhonesMYOB

Comment (joes-myob) [reply CHANGELOG.md]: I reverted these changes just in case

Comment (melboyce) [reply CHANGELOG.md]: https://github.com/myob-oss/hubr/blob/master/main.go#L1394
you'll also need to update comments and docs that refer to VERSION

**Pull request merged**